### PR TITLE
ompi: pml ucx: fix persistant request initialization

### PR DESCRIPTION
**Pull request by alex-mikheev**
Signed-off-by: Alex Mikheev <alexm@mellanox.com>
Fixes https://github.com/openucx/ucx/issues/1361
@yosefe

### CODE REVIEW

File: ompi/mca/pml/ucx/pml_ucx_request.c [hunk 1]
```diff
@@ -137,6 +137,12 @@ static void mca_pml_ucx_request_init_common(ompi_request_t* ompi_req,
     ompi_req->req_state            = state;
     ompi_req->req_free             = req_free;
     ompi_req->req_cancel           = req_cancel;
+    /* This field is used to attach persistant request to a temporary req.
+     * Receive (ucp_tag_recv_nb) may call completion callback
+     * before the field is set. If the field is not NULL then mca_pml_ucx_preq_completion() 
+     * will try to complete bogus persistant request.
+     */ 
+    ompi_req->req_complete_cb_data = NULL;
```
Comment (yosefe) [inline ompi/mca/pml/ucx/pml_ucx_request.c]: should we reset it to NULL after bsend completion?

Comment (alex-mikheev) [inline ompi/mca/pml/ucx/pml_ucx_request.c]: By default (on first allocation) request memory is zeroed ?

Comment (yosefe) [inline ompi/mca/pml/ucx/pml_ucx_request.c]: actually, now i see it is being reset on bsend completion.

**Comment by yosefe** [comment]
@alex-mikheev pls port to 2.x/2.0.x

**Pull request merged**